uat/testing-features/src/main/resources/greengrass/features/ggmq-1.feature

```diff
@@ -44,11 +44,11 @@ Feature: GGMQ-1
   }
 }
     """
-    And I connect device "test" on "agent1" to "default_broker" as "agent1-test-default_broker"
-    Then connection "agent1-test-default_broker" is successfully established within 3 seconds
-    When I subscribe "agent1-test-default_broker" to "iot_data_0" with qos 0
-    Then subscription to "iot_data_0" is successfull on "agent1-test-default_broker"
-    When I publish "agent1-test-default_broker" to "iot_data_0" with qos 0 and message "Test message"
-    Then publish message "test" to "iot_data_0" is successfully on "agent1-test-default_broker"
-    And message "Test message" received on "agent1-test-default_broker" from "iot_data_0" topic within 1 second
+    And I connect device "clientDeviceTest" on "agent1" to "default_broker" as "connection1"
+    Then connection "connection1" is successfully established within 3 seconds
```

Comment:
With this assumption we will miss all test cases when requires more than one connection for thingName. For example we can connnect one Thing to Iot Core directly and to local Greengrass. Another case create second connection to IoT core to check first connection will be broken. It is possible as test scenario. More production case: connect the same Thing to Greengrass1 and Greengrass2 as backup link,  I do not know allow IoT core how association of Thing with more that one Greengrass but can imagine such production scenaries. We know AWS has a millions of consumers with different needs, so we design control to be flexible as posible. Maybe we can assign Thing name to connectionName by default and simplify most of scenarios.

Reply:
You cannot have more than one connection per thing name. Thing name must be exactly equal to the client ID and client IDs must be unique. It is impossible to have more than one connection per thing.

Reply:
I know it is limitation of MQTT broker of IoT Core, but it may be not true for all brokers or for set of broker. At least we should be able to test that limitation of IoT core, right ?

Reply:
This test is for Greengrass, this is requirement of the Greengrass broker.

Reply:
The current implementation of test clients is flexible enough to establish several connections with different thing certificates to different brokers from single instance called "agent".
Do you want to hide this connection based logic or to make it optional? 
For example, after setup  "And I connect device "clientDeviceTest" on "agent1" to "default_broker" as "connection1"
we may execute same action using different syntax:
1. When I subscribe connection "connection1" ...
2. When I subscribe device "clientDeviceTest" ...
3. When I subscribe agent1 "agent1" ...
if there is only one connection then this steps will give us the same result

Reply:
@esmirnov-klika-tech I thing we can use thing name (exactly the same as clientId in terms of MQTT protocol) as connection name by default. It makes most of scenarios simplest. At the same time by design we will be able to write extended test scenarios by assign non-default connection name.

Reply:
Fixed 44986341adb2fd0b5bfbd49371c60b6ae4edc2f9